app/src/main/java/org/mozilla/fenix/onboarding/view/NotificationPermissionDialogScreen.kt

```diff
@@ -0,0 +1,221 @@
+                onClick = onDismiss,
+                modifier = Modifier.align(Alignment.End),
+            ) {
+                Icon(
+                    painter = painterResource(id = R.drawable.mozac_ic_close),
+                    contentDescription = stringResource(R.string.content_description_close_button),
+                    tint = FirefoxTheme.colors.iconPrimary,
+                )
+            }
+
+            NotificationPermissionPage(
+                pageState = notificationPermissionPageState,
+                onPrimaryButtonClick = onPrimaryButtonClick,
+                onSecondaryButtonClick = onSecondaryButtonClick,
+                modifier = Modifier
+                    .weight(1f)
+                    .fillMaxWidth()
+                    .padding(horizontal = 16.dp)
+                    .verticalScroll(rememberScrollState()),
+                imageModifier = Modifier
+                    .height(boxWithConstraintsScope.maxHeight.times(IMAGE_HEIGHT_RATIO)),
+            )
+        }
+    }
+}
+
+/**
+ * A page for displaying Notification Permission Content.
+ *
+ * @param pageState The page content that's displayed.
+ * @param onPrimaryButtonClick Invoked when the user clicks the primary button.
+ * @param onSecondaryButtonClick Invoked when the user clicks the secondary button.
+ * @param modifier The modifier to be applied to the Composable.
+ */
+@Composable
+private fun NotificationPermissionPage(
+    pageState: NotificationPermissionPageState,
+    onPrimaryButtonClick: () -> Unit,
+    onSecondaryButtonClick: () -> Unit,
+    modifier: Modifier = Modifier,
+    imageModifier: Modifier = Modifier,
+) {
+    Column(
+        modifier = modifier,
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.SpaceEvenly,
+    ) {
+        Image(
+            painter = painterResource(id = pageState.image),
+            contentDescription = null,
+            modifier = imageModifier,
+        )
+
+        Column(
+            modifier = Modifier.fillMaxWidth(),
+            horizontalAlignment = Alignment.CenterHorizontally,
+        ) {
+            Text(
+                text = stringResource(
+                    id = pageState.title,
+                    formatArgs = arrayOf(stringResource(R.string.app_name)),
+                ),
+                color = FirefoxTheme.colors.textPrimary,
+                textAlign = TextAlign.Center,
+                style = FirefoxTheme.typography.headline5,
+            )
+
+            Spacer(modifier = Modifier.height(16.dp))
+
+            Text(
+                text = stringResource(
+                    id = pageState.description,
+                    formatArgs = arrayOf(stringResource(R.string.app_name)),
+                ),
+                color = FirefoxTheme.colors.textSecondary,
+                textAlign = TextAlign.Center,
+                style = FirefoxTheme.typography.body2,
+            )
+        }
+
```

Comment:
While looking at the layout with @Kate-Galetski, we noticed an issue with small screen devices.   I suggest we add this [small_screen.patch](https://github.com/mozilla-mobile/fenix/files/10396409/small_screen.patch) so there is always a minimum 16dp between the text and the button.  This is approved by @Kate-Galetski.

Before(left)/After(Right)
![small_screen](https://user-images.githubusercontent.com/6579541/211932054-63572433-e12e-4e4c-b85d-d4297259cf96.jpg)

Reply:
Good catch, fixed [this](https://github.com/mozilla-mobile/fenix/pull/28477/files#diff-36208dfdd77d4e2cc25fa363acffad1873af086012f26eda6761f78c0e1ab112R180)!